osh/cmd_parse.py

```diff
@@ -1590,6 +1606,9 @@ def ParseOldCase(self, case_kw):
                     "This is a constant string.  You may want a variable like $x (parse_bare_word)",
                     loc.Word(w))
 
+        if w.tag() == word_e.Token:
```

Comment:
Let's write  != word_e.Compound, because we're expecting the CompoundWord for the case pattern


Those things are equivalent now, but they might not always be